requirements-training.txt

```diff
@@ -4,6 +4,6 @@ h5py
 numpy >= 1.16.6
 onnx
 packaging
-protobuf
```

Comment:
remove this change. we're taking care of it with https://github.com/microsoft/onnxruntime/pull/12132
the problem with pinning to 3.17.0 is anything between 3.12.2 and 3.20.1 is valid (from onnx perspective)

Reply:
Revoked the change @jywu-msft . Kindly review and merge the changes.